test_suite/package/otel_package/fixtures.py

```diff
@@ -0,0 +1,111 @@
+import json
+import time
+import pytest
+from lib import console_lib
+from lib import test_lib
+from test_suite.generic import helpers
+
+
+@pytest.fixture(scope='class')
+def initialize_veribales(request, host):
+    self = request.node.cls
+    instance_data = helpers.instance_data(host)
+    self.instance_id = instance_data['instance_id']
+    self.instance_dns = instance_data['public_dns']
+    self.instance_region = instance_data['availability_zone'][:-1]
+
+
+@pytest.fixture(scope='function')
+def check_instance_status(request, host, timeout_seconds=300):
```

Comment:
It's not only that, in additon I will have to declare the variable/s I want to use from this fixture here, for example:
self.instance_id = instance_data['instance_id']

Instead I'm doing this only **one time** in the "initialize_veribales" fixture that sets those variables to the class property which can be called by self anywhere in this code run. (including the test itself, and all the other fixtures within this run cause I'm using a request.node.cls that knows that this property belongs to Otel class.
In my opinion this solution is more appropriate for oop, but if you still think this is not a good idea let's talk about it and raise a better solution.

Reply:
The dicsion was to migrate __get_instance_data_from_json to helpers.py since it is more suitable as a general function rather configuration one before the tests, and to use it directly in "initialize_variables" fixture.